nodejs/week3/exercise.js

```diff
@@ -0,0 +1,135 @@
+    let createdDate = new Date();
+    let post = [id, title, band, venue, createdDate, performanceDate, price];
+    connection.query(`INSERT INTO concerts (id,title, band, venue, createdDate, performanceDate, price) VALUES (?,?,?,?,?,?,?)`,
+        post, (err, result) => {
+        if (err) throw err;
+      });
+}
+
+function showData(cb) {
+    connection.query("SELECT * FROM concerts", [], function (err, result) {
+        if (err) throw err;
+        cb(result);
+      });
+    
+}
+function showDataById(value, cb) {
+    connection.query("SELECT * FROM concerts WHERE id = ?", value, function (err, result) {
+        if (err) throw err;
+        cb(result);
+      });
+    
+}
+function showDataByCondition(condition, cb) {
+    let sql = connection.query("SELECT * FROM concerts WHERE ?? = ?", condition, function (err, result) {
+        if (err) throw err;
+        cb(result);
+      });
+}
+function showDateCreatedAfter(condition, cb) {
+    let sql = connection.query("SELECT * FROM concerts WHERE ?? > ?", condition, function (err, result) {
+        if (err) throw err;
+        cb(result);
+      });
+}
+function getLimit(value, cb) {
+    let sql = connection.query("SELECT * FROM concerts LIMIT ?", Number(value), function (err, result) {
+        if (err) throw err;
+        cb(result);
+      });
+}
+
+
+function deleteData(id, cb) {
+    connection.query("DELETE FROM concerts WHERE id = ?",id, function (err, result) {
+        if (err) throw err; 
+        cb(result);
+      });
+}
+function updateData(id, title, cb) {
+    connection.query("UPDATE concerts SET title = ? WHERE id = ?",[title,id], function (err, result) {
+        if (err) throw err;
+        cb(result);
+      });
+}
+
+// insertData(4, "abc","aabc","aaabc", new Date(),100);
+// insertData(5, "abcd","aabc","aaabc", new Date(),200);
+// deleteData(1);
+// updateData(4,"Blank title");
+
+app.get("/api/concerts", (request, response) => {
+    const priceValue = request.query.maxPrice;
+    const title = request.query.title;
+    const createdAfter = request.query.createdAfter;
+    const limit = request.query.limit;
+    let condition;
+
+    if(priceValue) {
+        condition = ["price", priceValue];
+        showDataByCondition(condition, (concert => response.send(concert)));
+    }
+    else if(title) {
+        condition = ["title", title];
+        showDataByCondition(condition, (concert => response.send(concert)));
+    }
+    else if(createdAfter) {
+        condition = ["createdDate", createdAfter];
+        showDateCreatedAfter(condition, (concert => response.send(concert)));
+    }
+    else if(limit) {
+        getLimit(limit, (concert => response.send(concert)));
+    }
+    else {
+        showData(concert => response.send(concert));
+    }
+});
+app.get("/api/concerts/:id", (request, response) => {
+    const id = request.params.id;
+    showDataById(id,(concert => response.send(concert)));
+});
+app.post('/api/concerts', (request, response) => {

```

Comment:
either use single semi-columns or double :) If you setup es-linter it will warn you against this too